Core/src/main/java/com/plotsquared/core/generator/HybridGen.java

```diff
@@ -377,6 +383,35 @@ public void initialize(PlotArea area) {
         // All initialization is done in the PlotArea class
     }
 
+    @Override
+    public BiomeType getBiome(final PlotArea settings, final int worldX, final int worldY, final int worldZ) {
+        HybridPlotWorld hybridPlotWorld = (HybridPlotWorld) settings;
+        if (!hybridPlotWorld.PLOT_SCHEMATIC && !hybridPlotWorld.ROAD_SCHEMATIC_ENABLED) {
+            return hybridPlotWorld.getPlotBiome();
+        }
+        int relativeX = worldX;
+        int relativeZ = worldZ;
+        if (hybridPlotWorld.ROAD_OFFSET_X != 0) {
+            relativeX -= hybridPlotWorld.ROAD_OFFSET_X;
+        }
+        if (hybridPlotWorld.ROAD_OFFSET_Z != 0) {
+            relativeZ -= hybridPlotWorld.ROAD_OFFSET_Z;
+        }
+        int size = hybridPlotWorld.PLOT_WIDTH + hybridPlotWorld.ROAD_WIDTH;
+        if (relativeX < 0) {
+            relativeX = size + (relativeX % size);
```

Comment:
This can probably be replaced with `Math.floorMod`, this way we don't need the if else